### PR TITLE
feat(policy): DSP-51 - deprecate PublicKey local field

### PR DESCRIPTION
**Pull request by suchak1**
resolves: https://github.com/opentdf/platform/issues/1358

### CODE REVIEW

File: service/policy/objects.proto [hunk 1]
```diff
@@ -308,11 +308,6 @@ message PublicKey {
       expression: "this.matches('^https://[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?(\\\\.[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?)*(/.*)?$')"
     }];
 
-    // public key; PEM of RSA public key; prefer `cached`
```
Comment (dmihalcik-virtru) [inline service/policy/objects.proto:311]: You'll need to reserve this number:

https://protobuf.dev/programming-guides/proto3/#reserved-field-numbers

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

**Comment by suchak1** [comment]
reserved fields are not supported in oneOf
https://github.com/protocolbuffers/protobuf/issues/5338
https://github.com/protobufjs/protobuf.js/issues/1008

File: service/policy/objects.proto (hunk 1 above)
Comment (suchak1) [reply service/policy/objects.proto:311]: @dmihalcik-virtru Could you take another look at this?

**Review by dmihalcik-virtru — DISMISSED** [review]

File: service/integration/attribute_fqns_test.go [hunk 2]
```diff
@@ -244,8 +244,14 @@ func (s *AttributeFqnSuite) TestGetAttributeByFqn_WithKeyAccessGrants_Definition
 	localKAS, err := s.db.PolicyClient.CreateKeyAccessServer(s.ctx, &kasregistry.CreateKeyAccessServerRequest{
 		Uri: "https://example.org/kas2",
 		PublicKey: &policy.PublicKey{
```
Comment (jakedoublev) [inline service/integration/attribute_fqns_test.go:246]: This looks great, @suchak1. Can you please add assertions to 1-2 tests in this suite and those in `kas_registry_test` that the key found in the KAS or Grant was the same as the 'cached' key used to create?

It looks like we had assertions around those for the local keys previously but none covering assertion that a KAS created with a cached key finds the same cached key on lookup (which tests our unmarshaling/marshaling of the proto to and from the db is sound).

Comment (suchak1) [reply service/integration/attribute_fqns_test.go:247]: added assertions to both test files

**Review by jakedoublev — APPROVED** [review]

**Review by strantalis — APPROVED** [review]

**Pull request merged**